daffodil-io/src/main/scala/edu/illinois/ncsa/daffodil/processors/charset/BitsCharset.scala

```diff
@@ -0,0 +1,243 @@
+package edu.illinois.ncsa.daffodil.processors.charset
+
+import edu.illinois.ncsa.daffodil.schema.annotation.props.gen.BitOrder
+import edu.illinois.ncsa.daffodil.exceptions.Assert
+import java.nio.charset.CoderResult
+import java.nio.charset.Charset
+import java.nio.charset.CodingErrorAction
+import java.nio.charset.CharsetEncoder
+import java.nio.charset.CharsetDecoder
+import java.nio.CharBuffer
+import java.nio.ByteBuffer
+import edu.illinois.ncsa.daffodil.util.MaybeULong
+import java.nio.charset.StandardCharsets
+
+/**
+ * Charset enhanced with features allowing it to work with Daffodil's Bit-wise
+ * DataInputStream and DataOutputStream.
+ */
+trait BitsCharset {
```

Comment:
Probably BitsCharset and DFDLCharset can be merged
    together so that all BitsCharset are serializable.